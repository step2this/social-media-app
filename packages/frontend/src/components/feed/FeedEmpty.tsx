import React from 'react';

export const FeedEmpty: React.FC = () => {
  return (
    <div className="feed-empty" data-testid="feed-empty" role="status">
      <span className="feed-empty__icon" aria-hidden="true">ğŸ“­</span>
      <p>No posts yet! Follow some users to see their posts here.</p>
    </div>
  );
};
