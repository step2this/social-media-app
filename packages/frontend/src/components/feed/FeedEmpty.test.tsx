import { describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import { FeedEmpty } from './FeedEmpty';

describe('FeedEmpty', () => {
  it('renders empty state message', () => {
    render(<FeedEmpty />);
    expect(
      screen.getByText(/No posts yet! Follow some users to see their posts here./i)
    ).toBeInTheDocument();
  });

  it('displays emoji/icon', () => {
    const { container } = render(<FeedEmpty />);
    const icon = container.querySelector('.feed-empty__icon');
    expect(icon).toBeInTheDocument();
  });

  it('shows helpful text', () => {
    render(<FeedEmpty />);
    const helpText = screen.getByText(/follow some users/i);
    expect(helpText).toBeInTheDocument();
  });

  it('has correct CSS classes', () => {
    const { container } = render(<FeedEmpty />);
    const emptyDiv = container.querySelector('.feed-empty');
    expect(emptyDiv).toBeInTheDocument();
    expect(emptyDiv).toHaveClass('feed-empty');
  });

  it('matches snapshot', () => {
    const { container } = render(<FeedEmpty />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('has accessible semantics', () => {
    const { container } = render(<FeedEmpty />);
    const emptyDiv = container.querySelector('.feed-empty');
    expect(emptyDiv).toHaveAttribute('role', 'status');
  });

  it('container has test ID', () => {
    render(<FeedEmpty />);
    expect(screen.getByTestId('feed-empty')).toBeInTheDocument();
  });

  it('renders without crashing', () => {
    const { container } = render(<FeedEmpty />);
    expect(container).toBeInTheDocument();
  });
});
