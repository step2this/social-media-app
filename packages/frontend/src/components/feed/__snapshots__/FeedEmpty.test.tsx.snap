// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FeedEmpty > matches snapshot 1`] = `
<div
  class="feed-empty"
  data-testid="feed-empty"
  role="status"
>
  <span
    aria-hidden="true"
    class="feed-empty__icon"
  >
    ðŸ“­
  </span>
  <p>
    No posts yet! Follow some users to see their posts here.
  </p>
</div>
`;
